R/string_operation.R

```diff
@@ -48,20 +48,20 @@ word_to_sentiment <- function(words, lexicon="bing"){
   # chosen lexicon and sentiment in words
   check <- sentiments$lexicon == lexicon
   if(lexicon == "nrc"){
-    sentiments <- (
-      sentiments[check,]
-      %>%  dplyr::group_by(word)
-      %>%  dplyr::summarize(sentiment=list(sentiment))
-      )
-  } else {
-    sentiments <- sentiments[check,]
-  }
-  joined_df <- dplyr::left_join(data.frame(word=words, stringsAsFactors = FALSE), sentiments, by="word")
-  if(lexicon=="AFINN"){
-    joined_df$score
-  } else {
-    joined_df$sentiment
+    sentiment_df <- sentiments[check,] %>%
```

Comment:
Please add comment on why this group_by/summarize is needed only for nrc case, and what this is doing.